File: php/concepts/variables.php
<?php

echo "\n=== Variables and Data Types ===\n";

// Variable names start with a $ sign.
$name = "John Doe";       // string
$age = 25;                // integer
$height = 1.75;           // double (float)
$is_student = true;       // boolean

// Use gettype() to find the type of a variable at runtime.
// PHP_EOL is a constant for the correct end-of-line character for the system.
echo "Name: $name (Type: " . gettype($name) . ")" . PHP_EOL;
echo "Age: $age (Type: " . gettype($age) . ")" . PHP_EOL;
echo "Height: $height (Type: " . gettype($height) . ")" . PHP_EOL;
echo "Is Student: $is_student (Type: " . gettype($is_student) . ")" . PHP_EOL;

File: golang/concepts/variables.go
package main

import "fmt"

// Variables & Data Types
func demonstrateVariables() {
	fmt.Println("\n=== Variables and Data Types ===")

	// In Go, data types are explicitly defined.
	// Go distinguishes between integers (int) and floating-point (float64) numbers.
	var name string = "John Doe"

	// We can use the `:=` operator for short variable declaration,
	// where Go will infer the data type.
	age := 25
	height := 1.75
	isStudent := true

	// The '%T' format verb is used to print the type of a variable.
	fmt.Printf("Name: %s (%T)\n", name, name)
	fmt.Printf("Age: %d (%T)\n", age, age)
	fmt.Printf("Height: %f (%T)\n", height, height)
	fmt.Printf("Is Student: %t (%T)\n", isStudent, isStudent)
}
